src/support.h

```diff
@@ -29,16 +32,44 @@ class AreaSupport {
      */
     static void generateSupportAreas(SliceDataStorage& storage, unsigned int mesh_idx, unsigned int layer_count, std::vector<Polygons>& supportAreas);
 
+    /*!
+     * Generate support bottom areas for a given mesh.
+     *
+     * The areas are put in the slice data storage.
+     *
+     * \param storage Where to find the previously generated support areas and
+     * where to output the new support bottom areas.
+     * \param mesh The mesh to generate support for.
+     */
+    static void generateSupportBottom(SliceDataStorage& storage, const SliceMeshStorage& mesh);
 
+    /*!
+     * Generate support roof areas for a given mesh.
+     *
+     * The areas are put in the slice data storage.
+     *
+     * \param storage Where to find the previously generated support areas and
+     * where to output the new support roof areas.
+     * \param mesh The mesh to generate support roof for.
+     */
+    static void generateSupportRoof(SliceDataStorage& storage, const SliceMeshStorage& mesh);
 
     /*!
-     * Generate support skin areas and non-skin areas for a given mesh.
-     * 
-     * \param storage Output storage: support area + support skin area output
-     * \param mesh The mesh to generate support skins for.
-     * \param layer_count The number of layers in this mesh group.
+     * \brief Generate a single layer of support interface.
+     *
+     * The interface polygons are going to get filled with the actual support
+     * interface polygons, and this is subtracted from the support areas.
+     *
+     * \param support_areas The areas where support infill is going to be
+     * printed.
+     * \param mesh_outlines The outlines of the mesh above or below the layer
+     * we're generating interface for. These layers determine what areas are
+     * going to be filled with the interface.
+     * \param safety_offset An offset applied to the result to make sure
+     * everything can be printed.
+     * \param interface An output variable to put the resulting polygons in.
```

Comment:
You should supply it as output parameter like so: `\param[out] description`
It's also possible to mark a parameter like `\param[in,out]`.
I normally leave out the `\param[in]` because that is the default behaviour of a function; there's no need to mark it as (als0) being an output variable.